### PR TITLE
Update PR template

### PR DESCRIPTION
**Pull request by praseodym**
## What & why

Update PR template to ask more context and get reviewers up to speed more quickly. This was one of the points from the last retro.

## How to test

Read the changes, no code changes to test.

## Reviewer notes

The checklist is commented out because I think that while it's useful as a reminder, it is too verbose to always display in the PR. However, [e-KS does use a PR checklist](https://github.com/kiesraad/e-ks/blob/main/.github/pull_request_template.md).

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,7 +1,21 @@
...
-- What's the scope of the changes?
-- What did you test? Which edge cases did you explicitly take into account?
-- Are there things you want reviewers to definitely take a look at?
-- Are there specific people you want feedback from?
-- Do you have tips on how to test? (For example: data setup, triggering specific errors, etc.)
+## Checklist
+
+- [ ] Single, focused change: unrelated changes split into separate PRs
+- [ ] I've self-reviewed the diff
+- [ ] Formatter and linter pass locally
```
Comment (rnijveld) [inline .github/pull_request_template.md:18]: Are we worried about this? This will fail on CI anway, all the other points are things that cannot be tested for automatically, whereas this one can. Other than that this looks very good to me!

Comment (praseodym) [reply .github/pull_request_template.md:18]: It does waste time to wait for CI and it's pretty fast to check this locally, so that's why I think it's good to include it in the checklist

**Review by rnijveld — APPROVED** [review]

**Review by cikzh — APPROVED** [review]

**Review by Lionqueen94 — APPROVED** [review]

**Review by oliver3 — APPROVED** [review]
